### PR TITLE
Update SonarJS to version 10.3.2.22047

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Implements #4336

### CODE REVIEW

**Comment by georgii-borovinskikh-sonarsource** [comment]
Manually tested, but pending confirmation from JS team that this version is backwards compatible (as a precaution)

File: src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs [hunk 1]
```diff
@@ -148,7 +151,7 @@ private static bool ProcessResource(string fullResourceName)
                     // Note: this is a quick way of getting the size of the document. Serializing the doc to a string
                     // and checking the length takes much longer (around 25 seconds)
                     var docLength = doc.ContentStart.DocumentStart.GetOffsetToPosition(doc.ContentEnd.DocumentEnd);
-                    Console.WriteLine($"{jsonRuleInfo.FullRuleKey}: size = {docLength}");
+                    // Console.WriteLine($"{jsonRuleInfo.FullRuleKey}: size = {docLength}");
```
Comment (duncanp-sonar) [inline src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs:154]: Does this need to be commented out?

**Review by duncanp-sonar — COMMENTED** [review]
Looks good, just a question about the commented-out line in the test

File: src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs:154]: Forgot to revert it before the commit. TBH, this line is pretty useless and was preventing me from seeing the actual errors so at one point I was considering removing it. What do you think?

Comment (duncanp-sonar) [reply src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs:154]: If it's not helping, remove it.

**Pull request merged**